apps/api/src/providers/provider.regression.ts

```diff
@@ -1,23 +1,55 @@
+import { ChannelTypeEnum, EmailProviderIdEnum } from '@novu/shared';
 import { UserSession } from '@novu/testing';
 import { expect } from 'chai';
 
-import { createRegressionNotificationTemplate } from './helpers';
+import { checkProviderIntegration, createProviderIntegration, createRegressionNotificationTemplate } from './helpers';
+import { getProviderSecrets } from './secrets';
 
-const providerId = 'sendgrid';
+const providers = [EmailProviderIdEnum.SendGrid];
```

Comment:
I will keep working with one provider only and then we can slowly add them and check that the tests don't need touch ups.